src/scheduler.cc

```diff
@@ -9,23 +9,71 @@
 #include <dirent.h>
 #include <dlfcn.h>
 #include <link.h>
-#include "rccl/rccl.h"
+#include <string>
+#include <curl/curl.h>
+#include <nlohmann/json.hpp>
+
+#ifdef RCCL
+  #include "rccl/rccl.h"
+#else 
+  #include "nccl.h"
+#endif
 #include "parser.h"
 
 #define __hidden __attribute__ ((visibility("hidden")))
 
-#define MSCCL_SCHEDULER_NAME "github.com/microsoft/msccl-scheduler"
+#define MSCCL_SCHEDULER_NAME "github.com/Azure/msccl-scheduler"
 
 static const char* mscclAlgoDirEnv = "MSCCL_ALGO_DIR";
 static const char* mscclAlgoDefaultDir = "msccl-algorithms";
 extern "C" bool mscclUnitTestMode() __attribute__((__weak__));
 static const char* mscclUnitTestAlgoDefaultDir = "msccl-unit-test-algorithms";
-static const char* mscclAlgoShareDirPath = "share/rccl/msccl-algorithms";
-static const char* mscclUnitTestAlgoShareDirPath = "share/rccl/msccl-unit-test-algorithms";
+static const char* mscclAlgoShareDirPath = "share/msccl-scheduler/msccl-algorithms";
+static const char* mscclUnitTestAlgoShareDirPath = "..share/msccl-unit-test-algorithms";
+static const char* mscclAzureVMDetectionAgent = "http://169.254.169.254/metadata/instance?api-version=2019-06-04";
```

Comment:
It cannot auto-detect A100 40G NDv4. Shall we add a warning message for user when auto-detection fails to find a matching VM?

Reply:
added.